app/components/pdf-image-preview/pdf-image-preview.component.ts

```diff
@@ -0,0 +1,101 @@
+import angular from "angular";
+import { getDocument, GlobalWorkerOptions, PDFPageProxy } from "pdfjs-dist";
+// Set the pdfjsLib worker source manually - https://github.com/mozilla/pdf.js/issues/8305
+import pdfjsWorker from "pdfjs-dist/build/pdf.worker.entry";
+
+interface IPDFImagePreviewBindings {
+  pdf: string | URL;
+  scale?: number;
+  canvasWidth?: number;
+  canvasHeight?: number;
+}
+
+interface IPDFImagePreviewCtrl extends IPDFImagePreviewBindings {
+  $onInit: () => void;
+  getImagePreviewFromPDF: (pdf: string | URL) => void;
+  makeThumb: (page: PDFPageProxy) => void;
+}
+
+class PDFImagePreviewCtrl implements IPDFImagePreviewCtrl {
+  static $inject: string[] = ["$element", "$scope", "$window"];
```

Comment:
I've been putting the $inject on the bottom, but if you want to start this convention in this shared package then migrate the other 2-3 components.